internal/mtls/caprovider_secret.go

```diff
@@ -0,0 +1,121 @@
+package mtls
+
+import (
+	"context"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"fmt"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	CASecretName = "k4e-ca"
+	providerName = "secret"
+
+	caCertSecretKey     = "ca.key"
+	caCertCertKey       = "ca.crt"
+	clientCertSecretKey = "client.key"
+	clientCertCertKey   = "client.crt"
+
+	certOrganization       = "k4e-operator"
+	certRegisterCN         = "register"
+	certDefaultExpiration  = 1 // years
```

Comment:
Let's make this configurable.

Reply:
Using ENV variable?

Reply:
yes, ENV variable is the option to go. we can just define it via ENV but not necessarily expose it in the configmap.

Reply:
https://issues.redhat.com/browse/ECOPROJECT-433